validator/sawtooth_validator/journal/consensus/consensus.py

```diff
@@ -16,27 +16,40 @@
 
 
 class BlockPublisherInterface(object):
-    """Consensus objects provide the following services to the Journal:
-    1) Build candidate blocks ( this temporary until the block types are
-    combined into a single block type)
-    2) Check if it is time to claim the current candidate blocks.
-    3) Provide the data a signatures required for a block to be validated by
-    other consensus algorithms
-
+    """The BlockPublisher provides consensus services to the BlockPublisher.
+    1) Block initialization or error if it is not time to create a block.
+    2) Check if it is time to claim the candidate blocks.
+    3) Sign the block prior to publishing. Provide an opaque block of data
+    that will allow the BlockVerifier implementation of this consensus
+    algorithm to verify that this is a valid block.
     """
-    # TDB - read only access the block store
-    # TBD - Early or late transaction binding -- voting needs to be early????
+
+    @abstractmethod
+    def __init__(self, block_cache, state_view):
+        """Initialize the object, is passed (read-only) state access objects.
+            Args:
+                block_cache: Dict interface to the block cache. Any predecessor
+                block to blocks handed to this object will be present in this
+                dict.
+                state_view: A read only view of state for the last committed
+                block in the chain. For the block publisher this is the block
+                we are building on top of.
+            Returns:
+                none.
+        """
+        pass
 
     @abstractmethod
     def initialize_block(self, block):
```

Comment:
This probably needs to have a `state_view` for on-chain settings purpose, based on the docstring.  PoET wait timer sounds like it might have some configuration to it...